src/components/navbar/navbar.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react'
+import { Link } from 'react-router-dom'
+
+const navBar = () => {
+    const [categories, setCategories] = useState([])
+
+    // Fetch categories from the api 
+    useEffect(() => {
+        fetch('https://dummyjson.com/categories')
+        .then((response) => response.json())
+        .then((data) => {
+            setCategories(data)
+        })
+        .catch((error) => {
+            console.log('Error fetching categories', error)
+        })
+    }, [])
+    
+    return (
+        <nav className="navbar">
+            <div className="navbar-logo">         
+                <Link to="/">
+                    <img src="public/img/logo.png" alt="logo" />
+                </Link>
+            </div> 
+            <div className="navbar-search">
+                <input type="text" placeholder="Search for products..." />
+            </div>
+            <div className="navbar-categories">
+                <ul>
+                    {categories.map((category) => (
+                        <li key={category._id}>
+                            <Link to={`/category/${category.slug}`}>{category.name}</Link>
```

Comment:
Curious what is `slug` property. unless it's a typo?

Reply:
typo